status/condition_set.go

```diff
@@ -104,18 +104,20 @@ func (c ConditionSet) IsTrue(conditionTypes ...string) bool {
 func (c ConditionSet) Set(condition Condition) (modified bool) {
 	conditionType := condition.Type
 	var conditions []Condition
-	for _, c := range c.object.GetConditions() {
-		if c.Type != conditionType {
-			conditions = append(conditions, c)
+	for _, cond := range c.object.GetConditions() {
+		if cond.Type != conditionType {
+			conditions = append(conditions, cond)
 		} else {
-			// If we'd only update the LastTransitionTime, then return.
-			condition.LastTransitionTime = c.LastTransitionTime
-			if reflect.DeepEqual(condition, c) {
+			condition.LastTransitionTime = metav1.Now()
+			condition.ObservedGeneration = c.object.GetGeneration()
+			if condition.Status == cond.Status && cond.LastTransitionTime.UnixNano() > 0 {
```

Comment:
```suggestion
			if condition.Status == cond.Status && !cond.LastTransitionTime.IsZero() {
```

This is more standard